pages/index.js

```diff
@@ -1,41 +1,125 @@
 import useSWR from "swr";
 import styled from "styled-components";
 import TaskPreviewCard from "../components/TaskPreviewCard";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
 
 export default function TasksListPage() {
+  // Fetch task data using useSWR hook
   const { data, isLoading } = useSWR("/api/tasks", { fallbackData: [] });
 
-  const sortedTasks = data.sort((taskA, taskB) => {
-    const dateA = new Date(taskA.creation_date);
-    const dateB = new Date(taskB.creation_date);
+  // State variables
+  const [sortedTasks, setSortedTasks] = useState([]); // Array to store sorted tasks
+  const [sortType, setSortType] = useLocalStorageState(
+    "sortType",
+    "creation_date"
+  ); // Sort type state with local storage persistence
+  const [sortDirection, setSortDirection] = useState("asc"); // Sort direction state
 
-    return dateB - dateA;
-  });
+  // Function to convert priority string to a corresponding number
+  const priorityToNumber = (priority) => {
+    switch (priority) {
+      case "low":
+        return 1;
+      case "medium":
+        return 2;
+      case "high":
+        return 3;
+      default:
+        return 0;
+    }
+  };
+
+  // Function to toggle the sort direction between ascending and descending
+  const toggleSortDirection = () => {
+    setSortDirection(sortDirection === "asc" ? "desc" : "asc");
```

Comment:
Just a suggestion: Maybe write the full words for asc and desc in your code already. With that I think i would not need an extra comment to understand the function as the function name is already really easy to understand